framework/Models/PostProcessors/Validations/PPDSS.py

```diff
@@ -0,0 +1,454 @@
+        targetOmegaNorm[cnt2] = targetProcessAction*targetOmega
+      #
+      featureProcessTimeNormScaled = np.zeros((y_count,z_count))
+      featureOmegaNormScaled = np.zeros((y_count,z_count))
+      for cnt3 in range(y_count):
+        featureProcessTimeNormScaled[cnt3] = featureProcessTimeNorm[cnt3]/timeScalingRatio
+        featureOmegaNormScaled[cnt3] = featureOmegaNorm[cnt3]/scaleRatioBeta
+      newfeatureData = np.asarray([featureOmegaNormScaled,featureProcessTimeNormScaled,featureBeta])
+      newtargetData = np.asarray([targetOmegaNorm,targetD,targetBeta])
+      #------------------------------------------------------------------------------------------
+      if pivotTargetSize == pivotSize:
+        timeParameter = pivotTarget
+      else:
+        timeParameter = pivotFeature
+      outputDict = {}
+      distanceTotal = np.zeros((y_count,z_count))
+      sigma = np.zeros((y_count,z_count))
+      for metric in self.metrics:
+        name = "{}_{}_{}".format(metric.estimator.name, targ.split("|")[-1], feat.split("|")[-1])
+      output = metric.evaluate((newfeatureData,newtargetData), multiOutput='raw_values')
+      #print(output)
+      for cnt2 in range(y_count):
+          distanceSum = abs(np.sum(output[cnt2]))
+          sigmaSum = 0
+          for cnt3 in range(z_count):
+            distanceTotal[cnt2][cnt3] = distanceSum
+            sigmaSum += output[cnt2][cnt3]**2
+          for cnt3 in range(z_count):
+            sigma[cnt2][cnt3] = (1/(z_count-np.sum(NaN_count[cnt2]))*sigmaSum)**0.5
+      rlz = []
+      for cnt in range(y_count):
+        outputDict = {}
+        outputDict[name] = abs(np.atleast_1d(output[cnt]))
+        outputDict['pivot_parameter'] = timeParameter
+        outputDict['total_distance_'+nameTarg[1]+'_'+nameFeat[1]] = distanceTotal[cnt]
+        outputDict['feature_beta_'+nameTarg[1]+'_'+nameFeat[1]] = featureBeta[cnt]
+        outputDict['target_beta_'+nameTarg[1]+'_'+nameFeat[1]] = targetBeta[cnt]
+        outputDict['feature_omega_'+nameTarg[1]+'_'+nameFeat[1]] = featureOmegaNormScaled[cnt]
+        outputDict['target_omega_'+nameTarg[1]+'_'+nameFeat[1]] = targetOmegaNorm[cnt]
+        outputDict['feature_D_'+nameTarg[1]+'_'+nameFeat[1]] = featureD[cnt]
+        outputDict['target_D_'+nameTarg[1]+'_'+nameFeat[1]] = targetD[cnt]
+        outputDict['process_time_'+nameTarg[1]+'_'+nameFeat[1]] = newfeatureData[1][cnt]
+        outputDict['standard_deviation_'+nameTarg[1]+'_'+nameFeat[1]] = sigma[cnt]
+        #print(newfeatureData[1][cnt])
+        rlz.append(outputDict)
+      realization_array.append(rlz)
+    #---------------
+    for cnt in range(len(realization_array[0])):
+      out = {}
+      for cnt2 in range(len(realization_array)):
+        for key, val in realization_array[cnt2][cnt].items():
+          out[key] = val
+      realizations.append(out)
+    return realizations
+
+  def _getDataFromDatasets(self, datasets, var, names=None):
```

Comment:
Is it function used by other post-processors, like "Probabilistic"? If so, please move it to the Validation base class